### ISSUE TITLE
Ignoring updates for untracked resources via annotation results in  increasing reconcile activity and cpu uses

### ISSUE THREAD

**Issue by asiyani** [argoproj/argo-cd#21154]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

After adding `argocd.argoproj.io/ignore-resource-updates` annotations to all `pods`
and `jobs` managed by ArgoCD we noticed increased reconcile activity and cpu uses.

We are using _application in any namespace_ feature and each application manages
full namespace. we got about 15 ArgoCD applications but only 2-3 namespaces had this
increased activity and `kube-system` is one of them.

following config is included in our [argocd-cm](https://github.com/utilitywarehouse/system-manifests/blob/master/argocd/namespaced/argocd-cm-patch.yaml)
```yaml
resource.ignoreResourceUpdatesEnabled: "true"
resource.customizations.ignoreResourceUpdates.all: |
    jsonPointers:
        - /status

resource.compareoptions: |
    ignoreDifferencesOnResourceUpdates: true
```

**To Reproduce**

Instead of updating all deployments we added annotations directly to all pods and jobs
using `kyverno` mutation policy.

**Expected behavior**

we were expecting reduced reconcile activity and cpu uses due to ignoring status
changes of pods and jobs.

**Screenshots**
![image](https://github.com/user-attachments/assets/c8e469db-19a3-4856-80a3-e05effe8f82d)

![image](https://github.com/user-attachments/assets/03bf55a0-2f68-4cdb-8378-a5f2a7627951)

**Version**

```shell
argocd: v2.11.0+d3f33c0
  BuildDate: 2024-05-07T16:21:23Z
  GitCommit: d3f33c00197e7f1d16f2a73ce1aeced464b07175
  GitTreeState: clean
  GoVersion: go1.21.9
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.13.1+af54ef8
  BuildDate: 2024-11-20T15:58:16Z
  GitCommit: af54ef8db5adfa77a08d4d05b1318a2198084c22
  GitTreeState: clean
  GoVersion: go1.23.1
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v5.4.3 2024-07-19T16:40:33Z
  Helm Version: v3.15.4+gfa9efb0
  Kubectl Version: v0.31.0
  Jsonnet Version: v0.20.0
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by andrii-korotkov-verkada** [comment]
Can you enable debug logs for the application controller and share ones related to one of applications? The best would be to have a comparison with and without this updates ignore with the rest of setup being the same.

**Comment by agaudreault** [comment]
Make sure your mutation is not constantly mutating the resources. If it does, then an increase would appear in Argo CD. 

If too many resources have that annotations, then the processing that Argo CD is doing can cause a CPU increase, but the count of reconciliation should not increase. This might indicate that the Kubernetes resource changed.

**Comment by asiyani** [comment]
Sorry for the delay..
After checking debug logs we found, the increased reconcile activity was due to `kyverno` generating/updating `updaterequests` resource for every pod it was mutating. 

```
❯ kubectl -n kube-system get updaterequests.kyverno.io   | wc -l
179
```

To fix this issue we added that resource to `resource.exclusions` list. since then we do not see increase in reconcile activity or the cpu uses as a result of adding annotation. 

One thing to note that adding this annotation also didn't `decrease` the reconcile activity as we expect it to do. 

![image](https://github.com/user-attachments/assets/6f8de1ff-9c5e-408b-8fa5-cb586790a685)

This issue can be closed as there is no side effect, apologies for raising incorrect issue.

**Comment by agaudreault** [comment]
Yeah, pods spec changes are often caused by new pods creation/deletion or status / readiness gates changes that are either relevant or affect the health status which will cause a reconcile. Ignoring ReplicaSet usually have a bigger impact, but I am glad to know that ignoring all Pods does not affect CPU negatively :)
